service/wave/wave.go

```diff
@@ -0,0 +1,334 @@
+	"github.com/spotinst/spotinst-sdk-go/spotinst/util/uritemplates"
+)
+
+type Cluster struct {
+	ID                *string      `json:"id,omitempty"`
+	ClusterIdentifier *string      `json:"clusterIdentifier,omitempty"`
+	Environment       *Environment `json:"environment,omitempty"`
+	Config            *Config      `json:"config,omitempty"`
+	State             *string      `json:"state,omitempty"`
+
+	// Read-only fields.
+	CreatedAt *time.Time `json:"createdAt,omitempty"`
+	UpdatedAt *time.Time `json:"updatedAt,omitempty"`
+}
+
+type Environment struct {
+	OperatorVersion         *string `json:"operatorVersion,omitempty"`
+	CertManagerDeployed     *bool   `json:"certManagerDeployed,omitempty"`
+	K8sClusterProvisioned   *bool   `json:"k8sClusterProvisioned,omitempty"`
+	OceanClusterProvisioned *bool   `json:"oceanClusterProvisioned,omitempty"`
+	EnvironmentNamespace    *string `json:"environmentNamespace,omitempty"`
+	OceanClusterId          *string `json:"oceanClusterId,omitempty"`
+}
+
+type Config struct {
+	Components []*Component `json:"components,omitempty"`
+}
+
+type Component struct {
+	Uid             *string           `json:"uid,omitempty"`
+	Name            *string           `json:"name,omitempty"`
+	OperatorVersion *string           `json:"operatorVersion,omitempty"`
+	Version         *string           `json:"version,omitempty"`
+	Properties      map[string]string `json:"properties,omitempty"`
+	State           *string           `json:"state,omitempty"`
+}
+
+type SparkApplication struct {
+	ID                *string `json:"id,omitempty"`
+	ApplicationID     *string `json:"applicationId,omitempty"`
+	ClusterIdentifier *string `json:"clusterIdentifier,omitempty"`
+	Name              *string `json:"name,omitempty"`
+	Namespace         *string `json:"namespace,omitempty"`
+	Heritage          *string `json:"heritage,omitempty"`
+	ApplicationState  *string `json:"applicationState,omitempty"`
+
+	// Read-only fields.
+	CreatedAt *time.Time `json:"createdAt,omitempty"`
+	UpdatedAt *time.Time `json:"updatedAt,omitempty"`
+}
+
+type ListClustersInput struct {
+	ClusterIdentifier *string `json:"clusterIdentifier,omitempty"`
+	ClusterState      *string `json:"clusterState,omitempty"`
+}
+
+type ListClustersOutput struct {
+	Clusters []*Cluster `json:"clusters,omitempty"`
+}
+
+type ReadClusterInput struct {
+	ClusterID *string `json:"clusterId,omitempty"`
+}
+
+type ReadClusterOutput struct {
+	Cluster *Cluster `json:"cluster,omitempty"`
+}
+
+type DeleteClusterInput struct {
+	ClusterID         *string `json:"clusterId,omitempty"`
+	ShouldDeleteOcean *bool   `json:"shouldDeleteOcean,omitempty"`
+}
+
+type DeleteClusterOutput struct{}
+
+type ListSparkApplicationsInput struct {
+	ClusterIdentifier *string `json:"clusterIdentifier,omitempty"`
+	Name              *string `json:"name,omitempty"`
+	Namespace         *string `json:"namespace,omitempty"`
+	ApplicationId     *string `json:"applicationId,omitempty"`
```

Comment:
```suggestion
	ApplicationID     *string `json:"applicationId,omitempty"`
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)